lumicks/pylake/population/hmm.py

```diff
@@ -0,0 +1,361 @@
+import warnings
+from dataclasses import dataclass
+
+import numpy as np
+
+from .mixture import GaussianMixtureModel
+from ..channel import Slice, Continuous
+
+
+class HiddenMarkovModel:
+    """A Hidden Markov Model describing hidden state occupancy and state transitions of observed
+    time series data (force, fluorescence, etc.)
+
+    This model accepts a 1D array as training data. *The state parameters are sorted according
+    to state mean* in order to facilitate comparison of models with different number of states
+    or trained on different datasets.
+
+    .. warning::
+
+        This is early access alpha functionality. While usable, this has not yet been tested in a
+        large number of different scenarios. The API can still be subject to change without any
+        prior deprecation notice! If you use this functionality keep a close eye on the changelog
+        for any changes that may affect your analysis.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Data array used for model training.
+    n_states : int
+        The number of hidden states in the model.
+    tol : float
+        The tolerance for training convergence.
+    max_iter : int
+        The maximum number of iterations to perform.
+    """
+
+    def __init__(self, data, n_states, tol, max_iter):
+        self.n_states = n_states
+        self._model, self._exit_flag = baum_welch(
```

Comment:
I like the approach of separating the model that uses all the shorthand notation from the user API. I would probably move that model to the `detail` folder (similarly to how we've done for the FEC models) if it's not meant to be used by end users.